### ISSUE TITLE
linux-5.10.y: ia64 build broken since 5.10.221-rc1

### ISSUE THREAD

**Issue by johnny-mnemonic** [linux-ia64/linux-stable-rc#1]
https://github.com/linux-ia64/linux-stable-rc/actions/runs/9771252437/job/26974019958#step:8:3524:
```
[...]
CC [M]  drivers/pps/pps.o
drivers/firmware/efi/memmap.c:16:10: fatal error: asm/efi.h: No such file or directory
   16 | #include <asm/efi.h>
      |          ^~~~~~~~~~~
compilation terminated.
make[3]: *** [scripts/Makefile.build:286: drivers/firmware/efi/memmap.o] Error 1
make[3]: *** Waiting for unfinished jobs....
  CC [M]  drivers/pps/kapi.o
make[2]: *** [scripts/Makefile.build:503: drivers/firmware/efi] Error 2
make[1]: *** [scripts/Makefile.build:503: drivers/firmware] Error 2
make[1]: *** Waiting for unfinished jobs....
[...]
```

**Comment by johnny-mnemonic** [comment]
I suspect a relation to the following change:

https://github.com/gregkh/linux/commit/31e0721aeabde29371f624f56ce2f403508527a5#diff-fbfb78e650ae911babf6c78ae5b7537bb6a449d19d116f89b34eea93d83f31acR16

...which adds `#include <asm/efi.h>` to [drivers/firmware/efi/memmap.c](https://github.com/gregkh/linux/blob/linux-5.10.y/drivers/firmware/efi/memmap.c).

I don't yet know why we don't hit this issue with other kernel branches, as [the actual change in mainline](https://github.com/gregkh/linux/commit/fdc6d38d64a20c542b1867ebeb8dd03b98829336) is included in mainline since v6.2-rc1 and most likely also in other linux stable branches. Maybe something else "fixes" the issue but didn't make it to linux-5.10.y?

**Comment by johnny-mnemonic** [comment]
Indeed, looks like https://github.com/linux-ia64/linux-stable-rc/commit/8ff059b8531f3b98e14f0461859fc7cdd95823e4 which:
> Move[s] some EFI related declarations that are only referenced on IA64 to
a new asm/efi.h arch header.

...was included in linux-5.12.y and upwards, but not below. Let's see if that patch set fixes the build for linux-5.10.y locally...

**Comment by johnny-mnemonic** [comment]
> ...was included in linux-5.12.y and upwards, but not below. Let's see if that patch set fixes the build for linux-5.10.y locally...

Confirmed, https://github.com/linux-ia64/linux-stable-rc/commit/8ff059b8531f3b98e14f0461859fc7cdd95823e4 indeed fixes the build.

**Comment by johnny-mnemonic** [comment]
[GHA build fixed for now](https://github.com/linux-ia64/linux-stable-rc/actions/runs/9871144965/job/27258970494#step:8:3805) with https://github.com/linux-ia64/linux-stable-rc/commit/081c0cc7bf17e00611a7c0f65affb6c692185d99.

Waiting for fix in upstream before closing this issue.

**Comment by johnny-mnemonic** [comment]
Fixed with the addition of 7a4dc0576ff2a177f951e90de7fb40d8089d4d10. Hence extra patch for linux-5.10.y removed with ab8a697ed1d50f98aff5b998409fca81f53a3510 and issue closed.
